src/bridges/convex/ConvexStakingBridge.sol

```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Aztec.
+pragma solidity >=0.8.4;
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {BridgeBase} from "../base/BridgeBase.sol";
+import {AztecTypes} from "rollup-encoder/libraries/AztecTypes.sol";
+import {ICurvePool} from "../../interfaces/curve/ICurvePool.sol";
+import {ErrorLib} from "../base/ErrorLib.sol";
+import {IRollupProcessor} from "rollup-encoder/interfaces/IRollupProcessor.sol";
+import {IConvexBooster} from "../../interfaces/convex/IConvexBooster.sol";
+import {ICurveLpToken} from "../../interfaces/convex/ICurveLpToken.sol";
+import {ICurveRewards} from "../../interfaces/convex/ICurveRewards.sol";
+
+/**
+ @notice A DefiBridge that allows user to stake Curve LP tokens through Convex Finance and earn boosted CRV 
+ without locking them in for an extended period of time. Plus earning CVX and possibly other rewards. 
+ @notice Staked tokens can be withdrawn (unstaked) any time.
+ @dev Convex Finance mints pool specific Convex LP token, however, not for the staking user (the bridge) directly. 
+ Hence, a storage of interactions with the bridge had to be implemented to know how much Curve LP token 
+ was staked and which Convex LP token was minted to represent this staking action. 
+ Since the Convex LP token is not returned to the bridge, Rollup Processor resp., 
+ a virtual asset is used on output and an interaction `receipt` is created utilizing the virtual asset ID. 
+ The receipt is then used to withdraw the staked means when a virtual asset of the matching ID is provided on input. 
+ @dev Synchronous and stateful bridge that keeps track of deposit interactions with the bridge that are later used for withdrawal.
+ @author Vojtech Kaiser
+ */
+contract ConvexStakingBridge is BridgeBase {
+    using SafeERC20 for IConvexBooster;
+    using SafeERC20 for ICurveLpToken;
+
+    struct PoolInfo {
+        uint256 poolId;
+        address curveLpToken;
+        address convexToken;
+        address curveRewards;
+        bool exists;
```

Comment:
The `exists` value is a bit redundant since you could simply check if any of the struct's addresses are equal to `address(0)`. In this case removing it won't save that much gas since the value will get packed with `curveRewards` address but I would remove it anyway.

Reply:
Resolved